packages/subgraph/src/util.ts

```diff
@@ -29,3 +84,218 @@ export function loadOrCreateAccount(accountId: string): Account {
+  lpPosition.unrealizedOSQTHUnitCost = ZERO_BD;
+  lpPosition.unrealizedETHUnitCost = ZERO_BD;
+
+  lpPosition.realizedOSQTHUnitCost = ZERO_BD;
+  lpPosition.realizedETHUnitCost = ZERO_BD;
+  lpPosition.realizedOSQTHUnitGain = ZERO_BD;
+  lpPosition.realizedETHUnitGain = ZERO_BD;
+  lpPosition.realizedOSQTHAmount = ZERO_BD;
+  lpPosition.realizedETHAmount = ZERO_BD;
+  return lpPosition as LPPosition;
+}
+
+export function loadOrCreateLPPosition(userAddr: string): LPPosition {
+  let lpPosition = LPPosition.load(userAddr);
+  // if no position, create new entity
+  if (lpPosition == null) {
+    lpPosition = new LPPosition(userAddr);
+    lpPosition = initLPPosition(userAddr, lpPosition);
+  }
+  return lpPosition as LPPosition;
+}
+
+export function loadOrCreatePosition(userAddr: string): Position {
+  let position = Position.load(userAddr);
+  // if no position, create new entity
+  if (position == null) {
+    position = new Position(userAddr);
+    position = initPosition(userAddr, position);
+  }
+  return position as Position;
+}
+
+export function getETHUSDCPrice(): BigDecimal[] {
+  let usdcPool = Pool.load(USDC_WETH_POOL);
+  if (usdcPool == null) {
+    return [ZERO_BD, ZERO_BD, ZERO_BD];
+  }
+  let usdcPrices = sqrtPriceX96ToTokenPrices(
+    usdcPool.sqrtPrice,
+    TOKEN_DECIMALS_USDC,
+    TOKEN_DECIMALS_18
+  );
+
+  return [usdcPool.sqrtPrice.toBigDecimal(), usdcPrices[0], usdcPrices[1]];
+}
+
+export function getoSQTHETHPrice(): BigDecimal[] {
+  let osqthPool = Pool.load(OSQTH_WETH_POOL);
+  let usdcPrices = getETHUSDCPrice();
+
+  if (osqthPool == null) {
+    return [ZERO_BD, ZERO_BD, ZERO_BD, ZERO_BD];
+  }
+  let osqthPrices = sqrtPriceX96ToTokenPrices(
+    osqthPool.sqrtPrice,
+    TOKEN_DECIMALS_18,
+    TOKEN_DECIMALS_18
+  );
+
+  return [
+    osqthPool.sqrtPrice.toBigDecimal(),
+    osqthPrices[0],
+    osqthPrices[1],
+    osqthPrices[0].times(usdcPrices[1]),
+  ];
+}
+
+export function createTransactionHistory(
+  transactionType: string,
+  event: ethereum.Event
+): TransactionHistory {
+  let transactionHistory = new TransactionHistory(
+    `${event.transaction.hash.toHex()}-${transactionType}`
+  );
+
+  let osqthPrices = getoSQTHETHPrice();
+  let usdcPrices = getETHUSDCPrice();
+  transactionHistory.sender = event.transaction.from;
+  transactionHistory.timestamp = event.block.timestamp;
+  transactionHistory.transactionType = transactionType;
+  transactionHistory.oSqthAmount = BigInt.zero();
+  transactionHistory.ethAmount = BigInt.zero();
+  transactionHistory.ethUSDCSqrtPrice = BigInt.fromString(
+    usdcPrices[0].toString()
+  );
+  transactionHistory.ethPriceInUSD = usdcPrices[1];
+  transactionHistory.ethOSQTHSqrtPrice = BigInt.fromString(
+    osqthPrices[0].toString()
+  );
+  transactionHistory.oSqthPriceInETH = osqthPrices[1];
+
+  return transactionHistory;
+}
+
+// buy: amount > 0
+// sell amount < 0
+export function buyOrSellETH(userAddr: string, amount: BigDecimal) {
```

Comment:
<img width="938" alt="image" src="https://user-images.githubusercontent.com/26922518/173506742-b9f94856-8562-4913-8f96-0f02a176ebfa.png">
Will need return type void

Reply:
Yes, just updated and pushed